packages/fuels/tests/harness.rs

```diff
@@ -2314,3 +2315,53 @@ async fn test_contract_id_and_wallet_getters() {
         contract_id
     );
 }
+
+#[tokio::test]
+async fn test_connect_get_gas_used() -> anyhow::Result<()> {
+    abigen!(
+        MyContract,
+        "packages/fuels/tests/test_projects/contract_test/out/debug/contract_test-abi.json"
+    );
+
+    let num_wallets = 2;
+    let num_coins = 3;
+    let amount = 1000000000000;
+    let config = WalletsConfig::new(Some(num_wallets), Some(num_coins), Some(amount));
+
+    let wallets = launch_custom_provider_and_get_wallets(config, None).await;
+
+    let id = Contract::deploy(
+        "tests/test_projects/contract_test/out/debug/contract_test.bin",
+        wallets.get(0).unwrap(),
+        TxParameters::default(),
+        StorageConfiguration::default(),
+    )
+    .await?;
+
+    let mut contract_instance =
+        MyContractBuilder::new(id.to_string(), wallets.get(0).unwrap().clone()).build();
+
+    assert_eq!(
+        contract_instance._get_wallet().address(),
+        wallets.get(0).unwrap().address()
+    );
+
+    let gas_used = contract_instance
+        .initialize_counter(42)
+        .call()
+        .await?
+        .gas_used;
+
+    assert!(gas_used > 0);
+
+    let gas_used_two = contract_instance
+        .connect(wallets.get(1).unwrap().clone())
+        .initialize_counter(42)
+        .call()
+        .await?
+        .gas_used;
+
+    assert!(gas_used_two > 0);
```

Comment:
here, you're not actually testing something that would "prove" that the transaction did occur with the second wallet. You're just testing that it spent some gas. What you can do is probably use non-zero gas price, and check that the wallet_1 balance is unchanged, while wallet_2 balance is diminished (base asset balance I mean).